### DISCUSSION TITLE
Columns right after list isn't displayed properly for reveal.js slides

### DISCUSSION THREAD

**Discussion by snhansen** [quarto-dev/quarto-cli#9951] [Q&A] [answered]
### Description

Consider the following example:

	---
	format: revealjs
	---

	## Title

	Here's a list

	* Item 1
	* Item 2

	:::: {.columns}

	::: {.column width=50%}
	Left column
	:::

	::: {.column width=50%}
	Right column
	:::

	::::

When rendering this as a revealjs slide, the columns aren't created properly:
![image](https://github.com/quarto-dev/quarto-cli/assets/45770265/7d31997e-cd48-4e6f-9be7-629447ce0da5)

It works if there's text or a `<br>` after the list. Is it a bug or is there a workaround? This is on Windows 11 using Quarto v. 1.5.43.

**Comment by mcanouil** [comment, marked as answer]
Try with:


````
---
format: revealjs
---

## Title

::: {#top-content}
Here's a list

* Item 1
* Item 2
:::

:::: {.columns}

::: {.column width=50%}
Left column
:::

::: {.column width=50%}
Right column
:::

::::

**Reply by snhansen** [reply]
That works. Thanks for the fast response @mcanouil!

**Reply by cscheid** [reply]
Hm. I'm surprised that this is necessary, though.

**Reply by mcanouil** [reply]
busy today, but since it's all Pandoc syntax, it's worth checking Pandoc directly for this example (was planning to do that at some point).

**Reply by mcanouil** [reply]
FYI, below is the HTML produced by Pandoc and Quarto.

<table>
<tr><th>Pandoc</th><th>Quarto</th></tr>
<tr><td>

````html
<section id="title" class="slide level2">
  <h2>Title</h2>
  <p>Here’s a list</p>
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
  </ul>
  <div class="columns">
    <div class="column" style="width:50%;">
      <p>Left column</p>
    </div>
    <div class="column" style="width:50%;">
      <p>Right column</p>
    </div>
  </div>
</section>
````

</td><td>

````html
<section id="title" class="slide level2">
  <h2>Title</h2>
  <p>Here’s a list</p>
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
  </ul>
  <div class="columns">
    <div class="column" style="width:50%;">
      <p>Left column</p>
    </div>
    <div class="column" style="width:50%;">
      <p>Right column</p>
    </div>
  </div>
  <div class="quarto-auto-generated-content">
    <div class="footer footer-default">
    </div>
  </div>
</section>
````

</td></tr>
</table>

So it might be a Reveal.js issue.
@cderv What do you think?

**Reply by cderv** [reply]
I believe this our Quarto CSS here. 
https://github.com/quarto-dev/quarto-cli/blob/974eba9aff5db8a17e9b45b457c6db3bfc0e34a8/src/resources/formats/html/_quarto-rules.scss#L642-L655

The `display: initial` here could be something causing this. Removing seems to solve as `display: flex` will be used. 

I believe this has been done because Pandoc did this change
* https://github.com/jgm/pandoc/pull/8237

and it seems there were breakage report which lead JJ to revert to initial. 

We would need to find the other example that don't work if we remove this initial value

**Reply by mcanouil** [reply]
Should we open an issue ("Future")?

**Reply by cderv** [reply]
yes we probably should to remember

**Reply by mcanouil** [reply]
- #10073
